backend/notifications.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+
+
+def HandleNotification(firebase, notification_id, notification):
+  """Helper function to propogate a notification."""
+
+  # TODO: Send to ionic
+  if 'playerId' in notification:
+    players = [notification['playerId']]
+  elif 'groupId' in notification:
+    players = firebase.get('/groups/%s' % notification['groupId'],
+                           'players')
+  else:
+    logging.error('Notification %s does not have a playerId or a groupId!' % (
+        notification_id))
+    return
+
+  for player in players:
+    firebase.put('/players/%s/notifications' % player,
+                  notification_id, notification)
+
+
+def ExecuteNotifications(request, firebase):
+  """INTERNAL ONLY: Send the notifications.
+
+  This method will execute queued notifications and send them to apps and add
+  them to the player specific notification database URLs.
+  """
+  # Handle notifications, 20 at a time so we don't need to get the entire
+  # notification database.
+  get_params = {
+      'orderBy': '"sent"',
+      'limitToLast': 20
+  }
+  current_time = time.time()
+  while True:
+    updates = False
+    notifications = firebase.get('/notifications', None, params=get_params)
+    if not notifications:
+      return
+    logging.info(notifications)
```

Comment:
This output could use some context in the logs or drop it if it's for dev debugging only.